apps/rush-lib/src/cli/actions/InstallAction.ts

```diff
@@ -6,128 +6,87 @@ import * as os from 'os';
   private _noLinkParameter: CommandLineFlagParameter;
 
   constructor(parser: RushCommandLineParser) {
     super({
       actionName: 'install',
-      summary: 'Install NPM packages in the Rush "common" folder, as specified by your shrinkwrap file.',
-      documentation: 'Always run "rush install" whenever you: (1) clone a repo, or (2) pull new changes from source'
-      + ' control, or (3) edit any package.json file.  The "rush install" command installs NPM packages into your'
-      + ' Rush "common" folder, using the exact versions specified in your npm-shrinkwrap.json file.'
-      + ' It also makes sure these versions satisfy your dependencies; if not, it will ask you to run'
-      + ' "rush generate". If there is nothing to do, then "rush install" won\'t take any time.'
-      + ' Afterwards, it will run "rush link" to create symlinks for all your projects.',
+      summary: '',
+      documentation: '',
       parser
     });
   }
 
   protected onDefineParameters(): void {
-    this._cleanInstall = this.defineFlagParameter({
+    this._cleanParameter = this.defineFlagParameter({
       parameterLongName: '--clean',
       parameterShortName: '-c',
-      description: 'Deletes the common "node_modules" folder and NPM cache before installing.'
-        + ' Use this option if you suspect that your package folder has become corrupted.'
-        + ' This occurs sometimes due to bugs in the NPM tool, or if you upgraded your'
-        + ' Node.js engine.'
-    });
-    this._cleanInstallFull = this.defineFlagParameter({
-      parameterLongName: '--full-clean',
-      parameterShortName: '-C',
-      description: '(UNSAFE!) Similar to "--clean", but also deletes and reinstalls shared files'
-        + ' such as the NPM tool itself. This is a more aggressive fix that is NOT SAFE to run'
-        + ' regularly because it may cause other Rush or NPM processes to fail.'
+      description: ''
     });
-    this._bypassPolicy = this.defineFlagParameter({
+    this._bypassPolicyParameter = this.defineFlagParameter({
       parameterLongName: '--bypass-policy',
       description: 'Overrides "gitPolicy" enforcement (use honorably!)'
     });
     this._noLinkParameter = this.defineFlagParameter({
       parameterLongName: '--no-link',
-      description: 'Do not automatically run the "rush link" action after "rush install"'
+      description: ''
     });
   }
 
   protected run(): Promise<void> {
-    if (!this._bypassPolicy.value) {
-      if (!GitPolicy.check(this.rushConfiguration)) {
-        process.exit(1);
-        return Promise.resolve();
-      }
+    const stopwatch: Stopwatch = Stopwatch.start();
+
+    this.eventHooksManager.handle(Event.preRushInstall);
 
-      ApprovedPackagesChecker.rewriteConfigFiles(this.rushConfiguration);
+    const purgeManager: PurgeManager = new PurgeManager(this.rushConfiguration);
+    const installManager: InstallManager = new InstallManager(this.rushConfiguration, purgeManager);
```

Comment:
>InstallManager [](start = 26, length = 14)

Should this be called `UpdateManager` then?

Reply:
Technically yes, but I think that would be misleading to people looking at the larger code base.  (It sounds like it could be updating Rush itself or somesuch.)  The centerpiece of InstallManager is "pnpm install", so I think the name is practical.

---
In reply to: [187452090](https://github.com/Microsoft/web-build-tools/pull/646#discussion_r187452090) [](ancestors = 187452090)